tests/test_run_service.py

```diff
@@ -0,0 +1,475 @@
+# -*- coding: utf-8 -*-
+"""Test run_service.py script using pytest for reliable automation."""
+
+import re
+import sys
+import logging
+import pexpect
+import os
+import time
+import pytest
+import tempfile
+import shutil
+from datetime import datetime
+from pathlib import Path
+from typing import Optional
+from termcolor import colored
+from colorama import init
+from web3 import Web3
+from eth_account import Account
+import requests
+import docker
+from dotenv import load_dotenv
+from operate.constants import HEALTH_CHECK_URL
+
+
+# Initialize colorama and load environment
+init()
+load_dotenv()
+
+STARTUP_WAIT = 10
+SERVICE_INIT_WAIT = 60
+CONTAINER_STOP_WAIT = 20
+
+# Handle the distutils warning
+os.environ['SETUPTOOLS_USE_DISTUTILS'] = 'stdlib'
+
+def check_docker_status(logger: logging.Logger) -> bool:
+    """Check if Docker containers are running properly."""
+    max_retries = 3
+    retry_delay = 20
+    
+    for attempt in range(max_retries):
+        logger.info(f"Checking Docker status (attempt {attempt + 1}/{max_retries})")
+        try:
+            client = docker.from_env()
+            
+            # Check all containers, including stopped ones
+            all_containers = client.containers.list(all=True, filters={"name": "traderpearl"})
+            running_containers = client.containers.list(filters={"name": "traderpearl"})
+            
+            if not all_containers:
+                logger.error(f"No trader containers found (attempt {attempt + 1}/{max_retries})")
+                if attempt == max_retries - 1:
+                    return False
+                logger.info(f"Waiting {retry_delay} seconds before retry...")
+                time.sleep(retry_delay)
+                continue
+            
+            # Log status of all containers
+            for container in all_containers:
+                logger.info(f"Container {container.name} status: {container.status}")
+                
+                if container.status == "exited":
+                    # Get exit code
+                    inspect = client.api.inspect_container(container.id)
+                    exit_code = inspect['State']['ExitCode']
+                    logger.error(f"Container {container.name} exited with code {exit_code}")
+                    
+                    # Get last logs
+                    logs = container.logs(tail=50).decode('utf-8')
+                    logger.error(f"Container logs:\n{logs}")
+                    
+                    # Try to restart the container if it exited with 0
+                    if exit_code == 0:
+                        logger.info(f"Attempting to restart container {container.name}")
+                        container.start()
```

Comment:
why do we try to restart the container?

Reply:
removed this code block, it was just there to restart incase of any error while starting conatiners in CI

Reply:
any error should fail the test, that's the purpose of the test :)

Reply:
noted